src/core/PerfLog.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+ * PerfLog.h - Small performance logger
+ *
+ * Copyright (c) 2017 gi0e5b06
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include "PerfLog.h"
+
+#ifdef LMMS_DEBUG_PERFLOG
+
+QHash <QString,PerfLog::Entry> PerfLog::s_running;
+
+PerfLog::Entry::Entry()
+{
+	c = times(&t);
+	if (c == -1) { qFatal("PerfLogEntry: init failed"); }
```

Comment:
I don't think using `qFatal` is reasonable here or in [line 52](https://github.com/LMMS/lmms/pull/3974/files#diff-49dbac95254771e56e0aad57952a47f8R52). If `times` fails, we can't mesaure the time here, but it doesn't mean we need to crash LMMS.